.ci/prepare-settings.sh

```diff
@@ -0,0 +1,40 @@
+#!/bin/bash
+set -e
+
+mkdir -p .ci-temp
+cp .ci/release-settings.xml .ci-temp/
+
+checkForVariable() {
+  if [[ "$(declare -p "$1")" =~ ^declare\ -x ]]; then
+    echo "Error: Define $1 environment variable"
+    exit 1
+  fi
+
+checkForVariable "SONATYPE_USER"
+checkForVariable "SONATYPE_PWD"
+checkForVariable "SOURCEFORGE_USER"
+checkForVariable "SOURCEFORGE_PWD"
+checkForVariable "GPG_PASSPHRASE"
+checkForVariable "GPG_KEY"
+
+replace() {
+  sed -i "s/$1/$2/g" .ci-temp/release-settings.xml
+}
+
+replace SONATYPE_USER
+replace SONATYPE_PWD
+replace SOURCEFORGE_USER
+replace SOURCEFORGE_PWD
+replace GPG_PASSPHRASE
+replace GPG_KEY
+}
+
+file1="./.ci-temp/release-settings.xml"
+file2="~/.m2/settings.xml"
+
+if cmp -s "$file1" "$file2"; then
+  date=$(date +"%m%d%y")
+ mv ~/.m2/settings.xml ~/.m2/settings.xml.backup."${date}"
+fi
+
+cp .ci-temp/release-settings.xml ~/.m2/settings.xml
```

Comment:
you have variables for these files

Reply:
We have variables for these files but we can use this only if we run those in `Travis CI`.

eg: `MVN_SETTINGS=${TRAVIS_HOME}/.m2/settings.xml`

Reply:
we should not use CI specific variables, this script should be executable on any CI and any local.

Reply:
I think exact path might work better then variable - it is almost unlikely to change

Reply:
what I mean is that several lines above it is
```
TEMP_SETTING="./.ci-temp/release-settings.xml"
SETTING="~/.m2/settings.xml"
```

so this command can be just `cp $TEMP_SETTING $SETTING` and duplcation is avoided

Reply:
now I see it, thanks a lot.
done.